[AMDSMI] Added Primary Partition information to conceptual doc (#6701)
ROCm/rocm-systems: projects/amdsmi/docs/conceptual/partition.md

      └── ...
```

For management purposes, AMD SMI treats Processor 0 (XCP 0) as the device's primary partition.
The physical GPU can still be managed from the primary partition, and it has full visibility into

> device-level metrics. All other partitions are referred to as secondary partitions.

Reviewer: The middle sentence is a bit circular ("can still be managed from the primary partition" repeats the first sentence, and "still" implies a contrast that hasn't been set up). Suggest tightening and bolding the new terms to match the doc's style:

```suggestion
AMD SMI treats Processor 0 (XCP 0) as the device's **primary partition**. The primary
partition has full visibility into **both device-level and partition-level metrics** and can
manage the whole physical GPU. All other partitions are **secondary partitions**, scoped to
their own resources.
```

Minor (optional): this paragraph sits a little orphaned between the tree diagram and the **Practical implications** heading — it might read more naturally folded into the `Processor / logical device` definition above (where XCP 0 is first introduced). Also worth keeping it anchored to `PARTITION_ID 0` / XCP 0 so readers can map it to `amd-smi list` output.

>       └── ...

Reviewer: The diagram above still uses the generic `XCP 0 / logical GPU` labels. Since the new text distinguishes primary vs. secondary, updating it to match would help readers map the two at a glance.

```suggestion
System
 └── Socket 0  (physical GPU / OAM)      ← amdsmi_socket_handle
      ├── Processor 0  (XCP 0 / logical GPU / primary)  ← amdsmi_processor_handle: gpu_metrics + xcp_metrics (e.g. renderD128)
      ├── Processor 1  (XCP 1 / logical GPU / secondary)  ← amdsmi_processor_handle: xcp_metrics only         (e.g. renderD129)
      └── ...  (count depends on active partition mode)
 └── Socket 1  (physical GPU / OAM)
      └── ...
```

- Metrics such as socket power (`socket_power`, `average_socket_power`) are reported at
  the socket level and reflect the total physical GPU. Per-XCP metrics (clocks, utilization,
  violations) are reported at the processor handle level.
- The primary partition can report device-wide totals for some metrics. Secondary partitions can

>   only report metrics scoped to their own partition.

Reviewer: This implies primary and secondary report the *same* metrics with different scope. In practice the device-wide metrics source isn't present on secondary nodes at all — secondary partitions only have `xcp_metrics`. This is also brand-new info for the doc (`sysfs`/`gpu_metrics`/`xcp_metrics` aren't mentioned anywhere else), but it ties back nicely to the renderD128-vs-renderD129 node discussion later in the file. Suggest:

```suggestion
- Partition (XCP) and device-level metrics come from **separate sysfs sources**. The
  device-wide `gpu_metrics` node exists only on the **primary partition** (XCP 0, e.g.
  `renderD128/device/gpu_metrics`), so only the primary partition can report whole-GPU
  values such as board power. **Secondary partitions** expose only their own `xcp_metrics`
  (e.g. `renderD129/device/xcp/xcp_metrics`) and therefore report metrics scoped to that
  partition — the device-wide set is not present on their node.
```
